src/extension/importer/rest/src/test/java/org/geoserver/importer/rest/ImportControllerTest.java

```diff
@@ -404,4 +404,83 @@ public void testGetImportGeoJSON() throws Exception {
+        // properly checked
+        createH2DataStore("gs", "skunkworks");
+
+        String json =
+                "{"
+                        + "\"import\": { "
+                        + "\"targetWorkspace\": {"
+                        + "\"workspace\": {"
+                        + "\"name\": \"sf\""
+                        + "}"
+                        + "},"
+                        + "\"targetStore\": {"
+                        + "\"dataStore\": {"
+                        + "\"name\": \"skunkworks\""
+                        + "}"
+                        + "}"
+                        + "}"
+                        + "}";
+
+        MockHttpServletResponse resp =
+                postAsServletResponse(
+                        RestBaseController.ROOT_PATH + "/imports", json, "application/json");
+        assertEquals(201, resp.getStatus());
+        assertNotNull(resp.getHeader("Location"));
+
+        int id = lastId();
+        assertTrue(resp.getHeader("Location").endsWith("/imports/" + id));
+
+        ImportContext ctx = importer.getContext(id);
+        assertNotNull(ctx);
+        assertNotNull(ctx.getTargetWorkspace());
+        // check that the context got the right workspace
+        assertEquals("sf", ctx.getTargetWorkspace().getName());
+        assertNotNull(ctx.getTargetStore());
+        assertEquals("skunkworks", ctx.getTargetStore().getName());
+        resp = postAsServletResponse(RestBaseController.ROOT_PATH + "/imports/" + id, "");
+        assertEquals(204, resp.getStatus());
+
+        json =
+                "{"
+                        + "\"import\": { "
+                        + "\"targetWorkspace\": {"
+                        + "\"workspace\": {"
+                        + "\"name\": \"gs\""
+                        + "}"
+                        + "},"
+                        + "\"targetStore\": {"
+                        + "\"dataStore\": {"
+                        + "\"name\": \"skunkworks\""
+                        + "}"
+                        + "}"
+                        + "}"
+                        + "}";
+
+        resp =
+                postAsServletResponse(
+                        RestBaseController.ROOT_PATH + "/imports", json, "application/json");
+        assertEquals(201, resp.getStatus());
+        assertNotNull(resp.getHeader("Location"));
+
+        id = lastId();
+        assertTrue(resp.getHeader("Location").endsWith("/imports/" + id));
+
+        ctx = importer.getContext(id);
+        assertNotNull(ctx);
+        assertNotNull(ctx.getTargetWorkspace());
+        // check that the context got the right workspace
+        assertEquals("gs", ctx.getTargetWorkspace().getName());
+        assertNotNull(ctx.getTargetStore());
+        assertEquals("skunkworks", ctx.getTargetStore().getName());
+        resp = postAsServletResponse(RestBaseController.ROOT_PATH + "/imports/" + id, "");
+        assertEquals(204, resp.getStatus());
+        removeStore("gs", "skunkworks");
```

Comment:
Best to do that in a @before, or in a "finally", to make sure it's always run (even in case this test start failing, otherwise it might pollute the state and cause other failures, making it all harder to debug).